### PR TITLE
Add bioimage search utils and example notebook

### PR DESCRIPTION
**Pull request by NikolaosPapailiou**
Add bioimage search utils and example notebook

### CODE REVIEW

**Review by ktsitsi — COMMENTED** [review]
Really great. One question how can we validate the similarity? In this type of images is not profound. In a test notebook I tried while ago the related_images included a top result the query image too. Validating that the `query image` is similar with the `top 1 related image`.

**Comment by NikolaosPapailiou** [comment]
> Really great. One question how can we validate the similarity? In this type of images is not profound. In a test notebook I tried while ago the `related_images` included as a top result the `query` image too. Validating that the `query image` is similar with the `top 1 related image`.

In the example Notebook, I currently skip the top result as it should always be the query image selected. However, in actual use cases, the query image would be an image that is not in the dataset so this shouldn't be problem. About result validation, do you have anything else in mind? The result similarity is mostly based on the embedding selected. Maybe you can find a finetuned embedding model for biomedical images that could give better results.

File: apis/python/src/tiledb/vector_search/embeddings/huggingface_auto_image_embedding.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        embeddings = np.zeros((size, self.dim_num), dtype=np.float32)
+        for image_id in range(len(objects["image"])):
+            image_batch.append(
+                Image.fromarray(
+                    np.reshape(objects["image"][image_id], objects["shape"][image_id])
+                )
+            )
+            count += 1
+            if count >= self.batch_size:
+                print(image_id)
```
Comment (ktsitsi) [inline apis/python/src/tiledb/vector_search/embeddings/huggingface_auto_image_embedding.py:61]: Is this a leftover `print`?

**Review by ktsitsi — APPROVED** [review]
Misc

File: apis/python/src/tiledb/vector_search/embeddings/huggingface_auto_image_embedding.py (hunk 1 above)
Comment (NikolaosPapailiou) [reply apis/python/src/tiledb/vector_search/embeddings/huggingface_auto_image_embedding.py:61]: Removed

**Review by jparismorgan — APPROVED** [review]

**Pull request merged**